libsrc/c64/crt0.s

```diff
@@ -98,14 +85,51 @@ L2:     lda     zpsave,x
 
         rts
 
+
 ; ------------------------------------------------------------------------
-; Data
 
-.segment        "ZPSAVE"
+.segment        "INIT"
 
-zpsave: .res    zpspace
+initstart:
 
-.bss
+; Save the zero-page locations that we need.
+
+        ldx     #zpspace-1
+L1:     lda     sp,x
+        sta     zpsave,x
+        dex
+        bpl     L1
+
+; Clear the BSS data.
+
+        jsr     zerobss
+
+; Set up the stack.
+
+        lda     #<(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
+        ldx     #>(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
+        sta     sp
+        stx     sp+1            ; Set argument stack ptr
+
+; Call the module constructors.
+
+        jsr     initlib
+
+; Push the command-line arguments; and, call main().
+
+        jmp     callmain
+
+
+; ------------------------------------------------------------------------
+; Data
+
+.data
 
-spsave: .res    1
 mmusave:.res    1
+spsave: .res    1
```

Comment:
Those variables have had a roundabout journey.  They started in DATA; then, they were moved to BSS; and now, I have moved them back to DATA.  I did it because the code that stores into them can be smaller if it is done early, before INIT is moved off of BSS.